mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -365,7 +398,11 @@ class LinearRegression @Since("1.3.0") (@Since("1.3.0") override val uid: String
       new BreezeOWLQN[Int, BDV[Double]]($(maxIter), 10, effectiveL1RegFun, $(tol))
     }
 
-    val initialCoefficients = Vectors.zeros(numFeatures)
+    val initialCoefficients = if (isSet(initialModel)) {
+      $(initialModel).coefficients
```

Comment:
Here we only use ```coefficients```, not use ```intercept```. This is because the ```intercept``` was computed using closed form after the coefficients are converged.